pkg/controller/utils/component.go

```diff
@@ -446,9 +446,16 @@ func mergeState(desired client.Object, current runtime.Object) client.Object {
 		cj := current.(*batchv1.Job)
 		dj := desired.(*batchv1.Job)
 
-		// We're only comparing jobs based off of annotations for now so we can send a signal to recreate a job. Later
-		// we might want to have some better comparison of jobs so that a changed in the container spec would trigger
-		// a recreation of the job
```

Comment:
For what it's worth, I would have left a comment of some sort in place here - having this called out is really useful for anyone coming in the future who may be wondering why the Job isn't being recreated. We didn't "solve" the problem outlined here - we just changed it from "annotations" to "annotations + images + #containers" 

There are a bunch of other fields that we still don't handle.

Reply:
Yes I didn't fully think this through. Comment is still definitely useful for people less familiar with the operator. (like me :sweat_smile: )

PR to restore comment with minor edit: https://github.com/tigera/operator/pull/3770